### DISCUSSION TITLE
New DreamBooth update is giving complete different results using same U_NET 2e-6 values as before

### DISCUSSION THREAD

**Discussion by meunumerotim333** [TheLastBen/fast-stable-diffusion#1610] [Q&A]
Please, I'm kind of desperate to have this solved, I used DreamBooth everyday to create images for my instagram, and today after I train Mo-Di model, it's giving me completely different results. I'm all day testing different methods but I'm frustrated.

**I always trainining using u_net 2e-6 since last year, always used the value of 1000 using 10 pictures on 768 pixels using 1.5 model**

**The results was always like the person I was training was imported to the model style.. and not sharp and boring results like too much realistic.**

I'm uploading an exemple on a image using the same pictures as I trained successfully before, even on the pictures I used the seed, the tests without the seed was terrible, very sharp and very realistic.. and isn't the intention.

I don't know how to upload image here, so I've uploaded on imgur.
**image link:**
https://i.imgur.com/YfSnopC.png

I also tried to train using 650 steps instead of 1000 but still is very sharp and isn't blending to the model style like I'm used too.

Please, I just would like to train the model again like I'm used to. I'm kind of desperate because without that my instagram will be probably dead. I don't know how to solve this without help.

**detailed configuration:**
model: 1.5 (modi external model)
u_net: 2e-6
text encoder: 350
steps: 1000
pictures: 10 768x

**Comment by TheLastBen** [comment]
Did you use the exact same settings and base model in both results in the image you uploaded ?

**Reply by meunumerotim333** [reply]
Yes, exactly the same. I use everyday and train persons everyday. I also did another tests with another people and same configuration and all are giving pretty sharp and realistic results like is over trained. But even if I low the value keeps the sharpness. Isn't blending with the model style. I'm so used to do now that I just train one time and is been perfect.. but today that started to happen and I'm 6 hours training to fix I think and keep frustrating : /

**Reply by meunumerotim333** [reply]
Looks like 4 days ago someone else complained about the same thing.. another information that I'vent mentioned is I always create colab copies and just create new ones after sometime.. for some reason not even the copies is working. Tried with a copy from 10 days before.. copy from 4 days before.. and all of them is giving the same bad results. The same copy that earlier have given good results. For some reason copies was affected too just today or yesterday.

There's the possibility to roll back 2e-6? I remember the advice you gave me someday about using 1e-5 + 8e-6 resuming.. but those configs gave me the same bad results as is giving me now. I just would like to have the 2e-6 like it was before. It was perfect. I hope it could be fixed, unfortunately I don't have good graphics card to do on my machine. Anyway I should thank for all the hard work, without you all of this wasn't possible. Thank you for the hark work

**Reply by TheLastBen** [reply]
it is 2e-6 now

**Reply by meunumerotim333** [reply]
Yes, its 2e-6 but something happening that changed the code or something. The results isnt't the same anymore :( 
Can you be able to check it and fix? Maybe just reroll some dates

**Reply by Bullseye-StableDiffusion** [reply]
> it is 2e-6 now
The last known working commit by me is "e59c0fc - fix RAM problem for good, thanks to @Daviljoe193". Maybe you can revert everything to that commit to solve the problem. I tried for many times and yea, something from the last 2 days broke it. Same settings, same model.

**Reply by THX1139b** [reply]
> Looks like 4 days ago someone else complained about the same thing

That would be me I think. I'm still having the same problems. Today I spent hours on 10 images starting very low in training steps and then resuming training and increasing. I based the training on both default SD 1.5 and Analog Diffusion which I have had great results with before as a basis. Saving a couple of renders, I then resumed training and rolled the same images again based on PNG info from those saved images to give me the exact same image with the resume-trained model.

At too low training steps there is hardly any likeness. It looks photorealistic but it's a different person. Increasing a bit gives better and better likeness but before it ever reaches a great likeness, the face becomes fake looking. When a very good likeness is reached, there are distortions to the face. I also did a few attempts at training an object with the same results really. Eventually reaching a good shape understanding only when image is distorted.

Using 2e-6 as before for all of these tests and I realize that this is the default from when I had better results before. Perhaps I will do another test with 40+ images later, as I used to use a lot more images than I typically do nowadays.

I also want to express my gratitude to TheLastBen for all of this! I complain but all of his work has given me more happiness all in all, than frustration!

(Edited)

**Reply by meunumerotim333** [reply]
> > Looks like 4 days ago someone else complained about the same thing
> 
> That would be me I think. I'm still having the same problems. Today I spent hours on 10 images starting very low in training steps and then resuming training and increasing. I based the training on both default SD 1.5 and Analog Diffusion which I have had great results with before as a basis. Saving a couple of renders, I then resumed training and rolled the same images again based on PNG info from those saved images to give me the exact same image with the resume-trained model.
> 
> At too low training steps there is hardly any likeness. It looks photorealistic but it's a different person. Increasing a bit gives better and better likeness but before it ever reaches a great likeness, the face becomes fake looking. When a very good likeness is reached, there are distortions to the face. I also did a few attempts at training an object with the same results really. Eventually reaching a good shape understanding only when image is distorted.
> 
> Using 2e-6 as before for all of these tests and I realize that this is the default from when I had better results before. Perhaps I will do another test with 40+ images later, as I used to use a lot more images than I typically do nowadays.
> 
> I also want to express my gratitude to TheLastBen for all of this! I complain but all of his work has given me more happiness all in all, than frustration!
> 
> (Edited)

Have you tried this colab reverted by a member of community? Is an old colab notebook that work as before this problem start:
https://colab.research.google.com/github/Bullseye-StableDiffusion/fixed/blob/main/fast_DreamBooth_fixed.ipynb

for me is working very well as I was used to

**Reply by THX1139b** [reply]
> Have you tried this colab reverted by a member of community? Is an old colab notebook that work as before this problem start: https://colab.research.google.com/github/Bullseye-StableDiffusion/fixed/blob/main/fast_DreamBooth_fixed.ipynb
> 
> for me is working very well as I was used to

Well, I finally tried this alternative notebook, speending a full workday on testing it. I tried a set of 28 very good quality photos of myself both at 512x512 and also at 768x768. 2e-6 with captions. The results look like Odo from Star Trek DS9, if that means anything. I resumed training and ended up training one of the batches for a full 2000 steps more but there was no discernible difference whatsoever from the first training run. It did look like me however, just more like a plastic prosthetic version of my face from a bad movie than a real human.

**Reply by meunumerotim333** [reply]
> > Have you tried this colab reverted by a member of community? Is an old colab notebook that work as before this problem start: https://colab.research.google.com/github/Bullseye-StableDiffusion/fixed/blob/main/fast_DreamBooth_fixed.ipynb
> > for me is working very well as I was used to
> 
> Well, I finally tried this alternative notebook, speending a full workday on testing it. I tried a set of 28 very good quality photos of myself both at 512x512 and also at 768x768. 2e-6 with captions. The results look like Odo from Star Trek DS9, if that means anything. I resumed training and ended up training one of the batches for a full 2000 steps more but there was no discernible difference whatsoever from the first training run. It did look like me however, just more like a plastic prosthetic version of my face from a bad movie than a real human.

But you were used to use a configuration before? Because for me it's working just like I was used to. Maybe you need to use another U_Net Learning value to be more realistic. My style of training is to fit the style of the trained model style.. but isn't photorealistic. I think every model will need to test different values of training. The Text encoder is also important.

**Reply by THX1139b** [reply]
> But you were used to use a configuration before? Because for me it's working just like I was used to. Maybe you need to use another U_Net Learning value to be more realistic. My style of training is to fit the style of the trained model style.. but isn't photorealistic. I think every model will need to test different values of training. The Text encoder is also important.

I’ve frankly lost track of exactly how everything was before. Mostly I used the default values with minor tweaks. I believe my best results were with a learning rate around 80 x or 100 x of the number of photos. I usually used 350 steps for the text encoder. And I used large image sets for training. Then I started mainly working with black and white photos of historical figures, which limited my choices to maximum 20 images that I could hunt down where the person had similar age, facial hair style and things like that. Also great results on most of my first models.

I tend to test on photorealism because if you can make a photorealistic believable character then that face will also work great for illustrations and whatnot. I’ve been making a lot of vintage poster art type images in the style of early 20th century advertising.

When I started getting good results with Dreambooth there was a new feature just added that let you use class images of faces of men, women, whatever. I used a set of 200 high quality photos that I put together but then that option quickly disappeared. Also there were no captions back then and a few other settings here and there that have changed. So my problem now may not be about u_net learning steps at all.

**Reply by meunumerotim333** [reply]
> > But you were used to use a configuration before? Because for me it's working just like I was used to. Maybe you need to use another U_Net Learning value to be more realistic. My style of training is to fit the style of the trained model style.. but isn't photorealistic. I think every model will need to test different values of training. The Text encoder is also important.
> 
> I’ve frankly lost track of exactly how everything was before. Mostly I used the default values with minor tweaks. I believe my best results were with a learning rate around 80 x or 100 x of the number of photos. I usually used 350 steps for the text encoder. And I used large image sets for training. Then I started mainly working with black and white photos of historical figures, which limited my choices to maximum 20 images that I could hunt down where the person had similar age, facial hair style and things like that. Also great results on most of my first models.
> 
> I tend to test on photorealism because if you can make a photorealistic believable character then that face will also work great for illustrations and whatnot. I’ve been making a lot of vintage poster art type images in the style of early 20th century advertising.
> 
> When I started getting good results with Dreambooth there was a new feature just added that let you use class images of faces of men, women, whatever. I used a set of 200 high quality photos that I put together but then that option quickly disappeared. Also there were no captions back then and a few other settings here and there that have changed. So my problem now may not be about u_net learning steps at all.

You can right click on the colab copy and revert old versions. Try to select an older one and see if revovers the one you were used to use.

**Comment by Omenizer** [comment]
Same here, can't get anything except the input images weirdly deformed?

**Reply by Bullseye-StableDiffusion** [reply]
> Same here, can't get anything except the input images weirdly deformed?

Yes, because the latest update broke something.
The last known working commit by me is "e59c0fc - fix RAM problem for good, thanks to @Daviljoe193". Maybe Ben can revert anything to that commit to solve the problem. I tried for many times and yea, something from the last 2 days broke it. Same settings, same model.

**Reply by Bullseye-StableDiffusion** [reply]
> Maybe it has to do something with that? ![изображение](https://user-images.githubusercontent.com/125838284/220155336-6da2f22b-08ac-4f38-a783-85c32bbc6caa.png)

100%

**Reply by meunumerotim333** [reply]
how to upload images here? didn't find the option.

Oh, I've also trained a model now because I've seen the Ben did an update. But is still giving very bad results

**Comment by Omenizer** [comment]
Tried it today, no change. Empty prompts are giving me the instance images, after very short training.

**Reply by TheLastBen** [reply]
I'm working on a solution

**Reply by Omenizer** [reply]
Thanks! 🙏🏼🙏🏼🙏🏼

**Reply by Bullseye-StableDiffusion** [reply]
> Tried it today, no change. Empty prompts are giving me the instance images, after very short training.

Try this notebook: https://github.com/Bullseye-StableDiffusion/fixed/blob/main/fast_DreamBooth_fixed.ipynb
It seems that I fixed the problem by reverting everything as it was 2-3 days ago. Tested and working 100%.

**Comment by TheLastBen** [comment]
Fixed with the new default values, 100 steps textenc and 500 UNet

**Reply by Omenizer** [reply]
For how many images do I use these settings (I have 9) and do they work for 1.5 (does anyone ever use 2.x?)?

Doing a run now, will report back! Thanks for looking into this

**Reply by Omenizer** [reply]
![image](https://user-images.githubusercontent.com/119709270/220354242-867572fb-9336-479c-aa00-f669256d4fee.png)

**Reply by Omenizer** [reply]
Overfit, all people look like the input. Need to reduce further, running with 300/50(!!!) now, almost done.

![image](https://user-images.githubusercontent.com/119709270/220360451-6c5d1ba8-837d-4878-8f3c-a399fedde700.png)

**Reply by Omenizer** [reply]
Ok still slightly overfitted, what do I reduce? This is getting weird. At least I can prompt celebs and they only look 20% like the subject now, but "person" still gives me my input face

**Reply by Omenizer** [reply]
200/35 now 😳

For 9 images, no captions

**Reply by TheLastBen** [reply]
If you get over fitting, reduce the learning rate to 1e-6, and make sure when you resume the training to set the textenc to 0. Also use no more than 10 high quality pictures.

**Reply by Bullseye-StableDiffusion** [reply]
Tried the newest version and it still has inferior results. At least slightly better than before this update, but inferior compared with an older commit. Tested them myself. Same model, same images.

**Reply by Omenizer** [reply]
> If you get over fitting, reduce the learning rate to 1e-6, and make sure when you resume the training to set the textenc to 0. Also use no more than 10 high quality pictures.

Hmm I'm in a weird place now, celebs and "person" look partly like my subject, but the subject doesn't look fully like itself. So both over- and underfitted at the same time 🥳

What are the **_respective effects_** of unet and text enoder training - can anyone explain?

**Reply by Omenizer** [reply]
I was already at 1e-6 for the text encoder

**Reply by Omenizer** [reply]
> > I was already at 1e-6 for the text encoder
> 
> Just try this collab from @Bullseye-StableDiffusion https://colab.research.google.com/github/Bullseye-StableDiffusion/fixed/blob/main/fast_DreamBooth_fixed.ipynb and with settings you usually used

dude thanks but you already told me twice

**Reply by meunumerotim333** [reply]
> > I was already at 1e-6 for the text encoder
> 
> Just try this collab from @Bullseye-StableDiffusion https://colab.research.google.com/github/Bullseye-StableDiffusion/fixed/blob/main/fast_DreamBooth_fixed.ipynb and with settings you usually used

thaaank you! My hero! Old method is just way better. Really important to still have this option. Many thanks

**Reply by Bullseye-StableDiffusion** [reply]
> > > I was already at 1e-6 for the text encoder
> > 
> > 
> > Just try this collab from @Bullseye-StableDiffusion https://colab.research.google.com/github/Bullseye-StableDiffusion/fixed/blob/main/fast_DreamBooth_fixed.ipynb and with settings you usually used
> 
> thaaank you! My hero! Old method is just way better. Really important to still have this option. Many thanks

Thanks for confirming that it works! And even better!

**Comment by Omenizer** [comment]
Anyone know that the Text Encoder influences, and what the effects of the Unet are? Or, what do I reduce/increase when?

**Reply by TheLastBen** [reply]
Train the text encoder only once for 80-100 steps, then in case you resume the training, set it to 0.

**Reply by Omenizer** [reply]
I'm at 35 steps now... What does the text encoder do exactly?

**Reply by TheLastBen** [reply]
it adds the new instance to the vocabulary (sort of)

**Comment by Omenizer** [comment]
Ok, I'm giving up for now. No idea what happened to the notebook - it was easy to get good results in 10 to 20 minutes, now I can't get anything usable at all :(

**Reply by meunumerotim333** [reply]
use the old colab notebook that our friend shared here above

this one https://colab.research.google.com/github/Bullseye-StableDiffusion/fixed/blob/main/fast_DreamBooth_fixed.ipynb
